src/main/java/com/microsoft/graph/http/IHttpRequest.java

```diff
@@ -154,5 +158,37 @@
      * @return Delay between retries in seconds
      */
     long getDelay();
+
+    /**
+     * Sets the HTTP method and returns the current request
+     *
+     * @param httpMethod the HTTP method
+     * @return the current request
+     */
+    IHttpRequest withHttpMethod(final HttpMethod httpMethod);
+    
+    /**
+     * Returns the Request object to be executed
+     * @return the Request object to be executed
+     */
+    Request getHttpRequest() throws ClientException;
+
+    /**
+     * Returns the Request object to be executed
+     * @param serializedObject the object to serialize at the body of the request
+     * @param <requestBodyType> the type of the serialized object
+     * @return the Request object to be executed
+     */
+    <requestBodyType> Request getHttpRequest(final requestBodyType serializedObject) throws ClientException;
+
+    /**
+     * Returns the Request object to be executed
+     * @param serializedObject the object to serialize at the body of the request
+     * @param progress the progress callback
+     * @param <requestBodyType> the type of the serialized object
+     * @param <responseType> the type of the response object
+     * @return the Request object to be executed
+     */
+    <requestBodyType, responseType> Request getHttpRequest(final requestBodyType serializedObject, final IProgressCallback<responseType> progress) throws ClientException;
```

Comment:
IHttpProvider should not contain an implementation specific type and should be generic instead